open_instruct/grpo_fast.py

```diff
@@ -1344,7 +1344,13 @@ def data_preparation_thread(
                     return  # Simply return to exit the thread cleanly
                 try:
                     result, batch = accumulate_inference_batches(
-                        inference_results_Q, pending_queries_map, args, training_step, generation_config, timeout=1.0
+                        inference_results_Q,
+                        pending_queries_map,
+                        args,
+                        training_step,
+                        generation_config,
+                        # We only timeout after half an hour. This is to avoid jobs hanging forever.
+                        timeout=1800.0,
```

Comment:
Can we set this to 1-2 hours, or make it configurable? With super-long contexts I might have jobs beyond 30minutes per rollout.

Reply:
I made it configurable! Testing now.

Reply:
Tests are good.